### DISCUSSION TITLE
Inpainting, unwanted changes at mask border

### DISCUSSION THREAD

**Discussion by leitgg** [lllyasviel/Fooocus#2769] [Q&A] [answered]
Hello community!

I use Fooocus inpainting to change backgrounds for a certain object.
For example, i take this image of a bottle:
![oel_flasche_klein](https://github.com/lllyasviel/Fooocus/assets/104002683/86b961ab-c4cc-4f50-a2f2-8b2c0c4ef5cb)

Then i remove the background with some tool and use the whole area around the bottle as mask to create a new background with Fooocus inpainting.
With some objects it works very well. But with ohter objects i never get a good result, meaning the object gets extended into the background like this:

![output_klein](https://github.com/lllyasviel/Fooocus/assets/104002683/c4f21a4e-5a21-4f33-8e48-f4771816dc4b)

Like the cap is wider.

Is there any way to avoid this?

I just want a new background and the bottle should keep its shape.

**Comment by Rogala** [comment]
It is desirable to adjust the size of the source image to one of the formats.
![345](https://github.com/lllyasviel/Fooocus/assets/7437582/5f6fafb5-d34c-40e4-8d11-c10efa187c23)

**Comment by mashb1t** [comment, marked as answer]
@leitgg this is a general issue of all AI tools and depends on how the AI handles the inpainting model.
It can be mitigated a little by eroding the inpaint mask foe a few pixels, but this still does not provide a perfect solution most of the time.

**Reply by leitgg** [reply]
Eroding the mask a bit gives indeed better results. As you say, not in all cases but it is better than before.

**Reply by zhaoqi571436204** [reply]
Can the fusion be carried out without eroding the content of the mask?

**Reply by mashb1t** [reply]
sure but as mentioned it is maybe not going to work as well
